source/fab/steps/compile_fortran.py

```diff
@@ -193,142 +184,89 @@ def store_artefacts(self, compiled_files: Dict[Path, CompiledFile], build_trees:
             new_objects = [lookup[af.fpath].output_fpath for af in source_files]
             object_files[root].update(new_objects)
 
-    # todo: identical to the c version - make a super class
     def process_file(self, analysed_file: AnalysedFile):
         """
         Prepare to compile a fortran file, and compile it if anything has changed since it was last compiled.
 
-        Returns a compilation result, including various hashes from the time of compilation:
-
-        * hash of the source file
-        * hash of the compile flags
-        * hash of the module files on which we depend
+        Returns a compilation result, regardless of whether it was compiled or prebuilt.
 
         """
-        # flags for this file
+        # todo: include compiler version in hashes
+
+        # get a combo hash of things which matter to the mod files we define
+        if analysed_file.file_hash is None:
```

Comment:
These asserts were added during debugging. I think this error handling should instead become a `try` around the `sum`.